### PR TITLE
add todo

### PR DESCRIPTION
**Pull request by SergeyOdnosteblets**
https://user-images.githubusercontent.com/59088833/195639437-09c89c6c-fdd4-4892-a8a4-f1b96945ee54.mp4

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,13 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className={backGround ? 'App light' : 'App dark'}>
```
Comment (t-shevchuk) [inline src/App.tsx]: It'll be better to use consts for "Light", "Dark" for `backGround`
or if you want to have boolean variable, then smth like `isDarkBackground`

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,13 @@
...
+      <Todo setBackGround={setBackGround} backGround={backGround} />
```
Comment (t-shevchuk) [inline src/App.tsx]: `onBackgroundChange`

File: src/Components/Form/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { FormTypes } from '../../Types/FormTypes';
+
+import style from './Form.module.scss';
+
+export const Form: React.FC<FormTypes> = ({ setTodos, todos }) => {
+  const handleSubmit = (e: any) => {
```
Comment (t-shevchuk) [inline src/Components/Form/Form.tsx:7]: is `e`  React.FormEvent<HTMLInputElement>?

Comment (t-shevchuk) [reply src/Components/Form/Form.tsx:7]: try to avoid any type

File: src/Components/Todo/Todo.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  const onChangeBackGround = () => {
+    setBackGround(!backGround);
+  };
+
+  return (
+    <div className={style.todo}>
+      <div className={style.title}>
+        <div className={style.title__text}>Todo</div>
+        <button onClick={() => onChangeBackGround()}>
```
Comment (t-shevchuk) [inline src/Components/Todo/Todo.tsx]: `onClick={onChangeBackGround}`

File: src/Components/Todo/Todo.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          <img src={moon} alt="" className={style.icon} />
```
Comment (t-shevchuk) [inline src/Components/Todo/Todo.tsx]: is `sun` added in other PRs?

File: src/Components/TodoInfo/TodoInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import { TaskType } from '../../Types/TaskType';
+import { TodoListType } from '../../Types/TodoListType';
+
+export const TodoInfo: React.FC<TodoListType> = ({ todos, setTodos, setFilteredTodos }) => {
+  const delCompletedTasks = () => {
+    setTodos(todos.filter((item: TaskType) => item.completed === false));
+  };
+
+  const getActiveTasks = () => {
+    setFilteredTodos(todos.filter((item: TaskType) => item.completed === false));
```
Comment (t-shevchuk) [inline src/Components/TodoInfo/TodoInfo.tsx]: filter((item: TaskType) => !item.completed)

File: src/Components/TodoInfo/TodoInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  const getCompletedTasks = () => {
+    setFilteredTodos(todos.filter((item: TaskType) => item.completed === true));
```
Comment (t-shevchuk) [inline src/Components/TodoInfo/TodoInfo.tsx]: filter((item: TaskType) => !!item.completed)

File: src/Components/TodoInfo/TodoInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { TaskType } from '../../Types/TaskType';
+import { TodoListType } from '../../Types/TodoListType';
+
+export const TodoInfo: React.FC<TodoListType> = ({ todos, setTodos, setFilteredTodos }) => {
+  const delCompletedTasks = () => {
+    setTodos(todos.filter((item: TaskType) => item.completed === false));
```
Comment (t-shevchuk) [inline src/Components/TodoInfo/TodoInfo.tsx]: filter((item: TaskType) => !item.completed)

File: src/Components/TodoInfo/TodoInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    setFilteredTodos(todos.filter((item: TaskType) => item.completed === true));
+  };
+
+  const getAllTasks = () => {
+    setFilteredTodos(todos);
+  };
+
+  return (
+    <div>
+      <div>{todos.filter((item: TaskType) => item.completed === false).length} items left</div>
```
Comment (t-shevchuk) [inline src/Components/TodoInfo/TodoInfo.tsx]: same here

File: src/Components/TodoItem/TodoItem.module.scss.d.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export const item: string
+export const text: string
+
```
Comment (t-shevchuk) [inline src/Components/TodoItem/TodoItem.module.scss.d.ts]: redundant empty lines

File: src/Components/TodoItem/TodoItem.tsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import { TodoItemType } from '../../Types/TodoItemType';
+
+import style from './TodoItem.module.scss';
+import del from '../../Images/icon-cross.svg';
+
+export const TodoItem: React.FC<TodoItemType> = ({ todo, deleteTask, checkedTask }) => {
+  return (
+    <Reorder.Item value={todo} key={todo.id} className={style.item}>
+      <input type="checkbox" onChange={() => checkedTask(todo.id)} checked={todo.completed} />
+      <span className={todo.completed ? style.text : ''}>{todo.title}</span>
```
Comment (t-shevchuk) [inline src/Components/TodoItem/TodoItem.tsx]: className={todo.completed && style.text}

File: src/Components/TodoList/TodoList.module.scss.d.ts [hunk 12]
```diff
@@ -0,0 +1,4 @@
+export const todoList: string
+
+
```
Comment (t-shevchuk) [inline src/Components/TodoList/TodoList.module.scss.d.ts]: redundant empty lines

File: src/Components/TodoList/TodoList.tsx [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  filteredTodos,
+}) => {
+  const [refresh, setRefresh] = useState(false);
+  const deleteTask = (id: number) => {
+    setTodos(todos.filter((item: TaskType) => item.id !== id));
+  };
+
+  const checkedTask = (id: number) => {
+    todos.filter((item: TaskType) => {
+      if (item.id === id) {
```
Comment (t-shevchuk) [inline src/Components/TodoList/TodoList.tsx]: use splice here instead

File: src/Types/FormTypes.ts [hunk 14]
```diff
@@ -0,0 +1,5 @@
+export interface FormTypes {
+    todos: any
```
Comment (t-shevchuk) [inline src/Types/FormTypes.ts]: Why it's any type?

File: src/Types/FormTypes.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
...
+    setTodos: (event: any) => void
```
Comment (t-shevchuk) [inline src/Types/FormTypes.ts]: and here

File: src/Types/TodoItemType.ts [hunk 16]
```diff
@@ -0,0 +1,7 @@
+import { TaskType } from './TaskType';
+
+export interface TodoItemType {
+    todo: TaskType
+    deleteTask: (e: number) => void
```
Comment (t-shevchuk) [inline src/Types/TodoItemType.ts]: What is `e`?
And below

File: src/Types/TodoListType.ts [hunk 17]
```diff
@@ -0,0 +1,9 @@
+export interface TodoListType {
+    todos: any
```
Comment (t-shevchuk) [inline src/Types/TodoListType.ts]: why any?

File: src/Types/TodoListType.ts [hunk 18]
```diff
@@ -0,0 +1,9 @@
...
+    setTodos: (event: any) => void
+    setFilteredTodos:(event: any) => void
+    filteredTodos: any
+}
+
```
Comment (t-shevchuk) [inline src/Types/TodoListType.ts]: lines

File: src/Types/TodoType.ts [hunk 19]
```diff
@@ -0,0 +1,7 @@
+export interface TodoType {
+  backGround: boolean
+  setBackGround: (arg:boolean) => void
+}
+
```
Comment (t-shevchuk) [inline src/Types/TodoType.ts]: lines

File: src/Components/Todo/Todo.tsx (hunk 5 above)
Comment (SergeyOdnosteblets) [reply src/Components/Todo/Todo.tsx]: change

File: src/Types/TodoItemType.ts (hunk 16 above)
Comment (SergeyOdnosteblets) [reply src/Types/TodoItemType.ts]: it's id, changed

File: src/Types/TodoListType.ts (hunk 17 above)
Comment (SergeyOdnosteblets) [reply src/Types/TodoListType.ts]: changed it

File: src/Types/FormTypes.ts (hunk 14 above)
Comment (SergeyOdnosteblets) [reply src/Types/FormTypes.ts]: changed

File: src/Components/StatusItems/StatusItems.module.scss.d.ts [hunk 20]
```diff
@@ -0,0 +1,6 @@
+export const buttons: string
+export const button: string
+
```
Comment (t-shevchuk) [inline src/Components/StatusItems/StatusItems.module.scss.d.ts]: empty lines

File: src/Components/TodoInfo/TodoInfo.module.scss.d.ts [hunk 21]
```diff
@@ -0,0 +1,7 @@
+export const info: string
+export const button: string
+export const statusItems: string
+
```
Comment (t-shevchuk) [inline src/Components/TodoInfo/TodoInfo.module.scss.d.ts]: empty lines

File: src/Types/TodoInfoType.ts [hunk 22]
```diff
@@ -0,0 +1,14 @@
...
+export interface TodoInfoType {
+    todos: TaskType[]
+    setTodos: (event: TaskType[]) => void
+    filteredTodos: TaskType[]
+    onBackgroundChange: boolean
+    getActiveTasks: () => void
+    getCompletedTasks: () => void
+    getAllTasks: () => void
+}
+
```
Comment (t-shevchuk) [inline src/Types/TodoInfoType.ts]: empty lines

**Review by t-shevchuk — APPROVED** [review]

**Pull request merged**